index.d.ts

```diff
@@ -0,0 +1,3 @@
+interface TemplateFactory {}
+export function hbs(template: string): TemplateFactory;
+export function hbs(tagged: TemplateStringsArray): TemplateFactory;
```

Comment:
Hmm. The `TemplateFactory` definition here won't actually *do* anything; it's just an empty interface. We need to do one of:

- importing the *actual* type from Ember's types and using that so we can guarantee interop (recommended)
- making it return `unknown`, which will require casts for folks to use it (not recommended)